internal/gomod/dependencies.go

```diff
@@ -95,15 +120,88 @@ func parseGoListOutput(output, rootVersion string) (map[string]Module, error) {
 			module.Version = rootVersion
 		}
 
-		dependencies[importPath] = Module{
+		dependencies[importPath] = GoModule{
 			Name:    module.Name,
 			Version: cleanVersion(module.Version),
 		}
 	}
 
+	if goVersion.Version == "" {
+		return nil, errors.New("Must have a valid go version in go mod file")
+	}
+	setGolangDependency(dependencies, goVersion)
+
 	return dependencies, nil
 }
 
+type parsedGoVersion struct {
+	Version string
+	Major   int
+	Minor   int
+}
+
+func getGoVersion(version string) (parsedGoVersion, error) {
+	splitVersion := strings.Split(version, ".")
+
+	major, err := strconv.Atoi(splitVersion[0])
+	if err != nil {
+		return parsedGoVersion{}, err
+	}
+
+	minor, err := strconv.Atoi(splitVersion[1])
+	if err != nil {
+		return parsedGoVersion{}, err
+	}
+
+	return parsedGoVersion{
+		Version: version,
+		Major:   major,
+		Minor:   minor,
+	}, nil
+}
+
+func setGolangDependency(dependencies map[string]GoModule, goVersion parsedGoVersion) {
+	dependencies[golangRepository] = GoModule{
+		Name:    golangRepository,
+		Version: "go" + goVersion.Version,
+	}
+
+}
+
+func GetGolangDependency(dependencies map[string]GoModule) GoModule {
+	return dependencies[golangRepository]
+}
+
+// IsStandardlibPackge checks whether a particular package is a standard
+// library package.
+func IsStandardlibPackge(pkg string) bool {
+	if strings.Contains(pkg, ".") {
```

Comment:
Can we describe the heuristics here? "." is because it lacks a domain name in the import path.